core/server/worker/src/main/java/alluxio/worker/block/MonoBlockStore.java

```diff
@@ -301,46 +312,73 @@ public List<BlockStatus> load(List<Block> blocks, String tag, OptionalLong bandw
         blockWriter = mLocalBlockStore.createBlockWriter(sessionId, blockId);
       } catch (Exception e) {
         AlluxioRuntimeException exception = AlluxioRuntimeException.from(e);
-        errors.add(BlockStatus.newBuilder().setBlock(block)
-            .setCode(exception.getStatus().getCode().value()).setMessage(e.getMessage())
-            .setRetryable(exception.isRetryable()).build());
+        BlockStatus.Builder builder = BlockStatus.newBuilder().setBlock(block)
+            .setCode(exception.getStatus().getCode().value()).setRetryable(exception.isRetryable());
+        if (exception.getMessage() != null) {
+          builder.setMessage(exception.getMessage());
+        }
+        errors.add(builder.build());
+        if (hasTempBlockMeta(blockId)) {
+          try {
+            abortBlock(sessionId, blockId);
+          } catch (Exception ee) {
+            LOG.warn(format("fail to abort temp block %s when creating block for load request",
+                block.getBlockId()), ee);
+          }
+        }
         continue;
       }
-      CompletableFuture<BlockStatus> future = RetryUtils.retryCallable("read from ufs",
+      CompletableFuture<Void> future = RetryUtils.retryCallable("read from ufs",
               () -> manager.read(blockId, block.getOffsetInFile(), blockSize, block.getUfsPath(),
                   false, tag), new ExponentialBackoffRetry(1000, 5000, 5))
+          // use orTimeout in java 11
+          .applyToEither(timeoutAfter(LOAD_TIMEOUT, TimeUnit.MILLISECONDS), d -> d)
```

Comment:
Please add a unit test to test timeout behavior. (mock load API and make sure it triggers timeout)